graphics/transparent_surface.cpp

```diff
@@ -326,6 +327,72 @@ void doBlitSubtractiveBlend(byte *ino, byte *outo, uint32 width, uint32 height,
 	}
 }
 
+/**
+ * Optimized version of doBlit to be used with multiply blended blitting
+ */
+void doBlitMultiplyBlend(byte *ino, byte *outo, uint32 width, uint32 height, uint32 pitch, int32 inStep, int32 inoStep, uint32 color) {
+	byte *in;
+	byte *out;
+
+	if (color == 0xffffffff) {
+		for (uint32 i = 0; i < height; i++) {
+			out = outo;
+			in = ino;
+			for (uint32 j = 0; j < width; j++) {
+
+				if (in[kAIndex] != 0) {
+					out[kRIndex] = MIN((in[kRIndex] * in[kAIndex] >> 8) * out[kRIndex] / 225, 255);
```

Comment:
Also, I would challenge the statement that using a bit shift is better. I may be incorrect, but my understanding is that all compilers will perform peephole optimisation of unsigned division with a 2^n denominator to turn it into a bit shift (or whatever instruction is optimal on a given platform) automatically. However, whereas arithmetic division of negative numerators is well-defined, bit shifting negative integers is either UB (left shift) or implementation-defined (right shift). So you can get into UB territory more easily when bit shifting manually, without actually having any performance improvement since the compiler will optimise it anyway if it is possible.